lib/ftpWrapper.js

```diff
@@ -0,0 +1,66 @@
+var ftp = require('ftp');
+
+var clients = {};
+
+function connect(urlObj, callback, errback){
+
+  if(clients[urlObj.hostname]) return process.nextTick(function(){
```

Comment:
So that the calling function can always expect this to call the callback asynchronously. Otherwise any out of order setup in the caller that assumes asynchrony will fail.